File: Java/167.两数之和II-输入有序数组.java
class Solution {
    //由于已经有序，双指针遍历即可
    public int[] twoSum(int[] numbers, int target) {
        int[] res = new int[2];
        int n = numbers.length;
        if(n < 2) return res;
        int i = 0, j = n - 1;
        while(i < j){
            int sum = numbers[i] + numbers[j];
            if(sum == target){
                res[0] = i + 1;
                res[1] = j + 1;
                return res;
            }else if(sum > target){
                j--;
            }else{
                i++;
            }
        }
        return res;
    }
}

File: CPP/167. 两数之和 II - 输入有序数组.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        vector<int> res = {-1,-1};
        if(numbers.empty() || numbers.size() < 2) return res;
        int left = 0, right = numbers.size()-1;
        while(left < right){
            if(numbers[left] + numbers[right] == target){
                res[0] = left + 1;
                res[1] = right + 1;
                return res;
            }else if(numbers[left] + numbers[right] > target){
                right--;
            }else{
                left++;
            }
        }
        return res;
    }
};